ElevatorSim/src/elevatorsim/scheduler/Scheduler.java

```diff
@@ -47,8 +46,8 @@ public static Scheduler getInstance() {
 	 * Adds a floor to this scheduler to manage
 	 * @param floor - the floor to add
 	 */
-	public void addFloor(Floor floor) {
-		floors.add(floor);
+	public void setFloorController(FloorController floorSystem) {
```

Comment:
I am considering a factory pattern to prevent people from changing floor controller or elevators after the thread starts.

It would be a weird factory that builds a singleton... does a pattern like that even exist?

Reply:
How does implementing the factory pattern ensure that people don't modify it? Should we not just make the elevator controller and elevators Singleton's if the idea is to just create a single instance for each object?

Reply:
My thoughts were to remove the setFloorController and addElevators to the factory. Without those it won't be possible to modify it. This works for now, just a thought I had.

Reply:
Ah okay I think I get what you want to do. I think that should be saved for the next iteration for now.